### DISCUSSION TITLE
Welcome To The Telemetrix Discussion Page

### DISCUSSION THREAD

**Discussion by MrYsLab** [MrYsLab/telemetrix#10] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. Use the discussions are to:
  * Ask questions you’re wondering about and are not issue specific.
  * Share ideas.
  * Engage with other community members.

   Select the appropriate category and enter your discussion post!

**Comment by PG-InGitHub** [comment]
How to control LCD1602 and nixie tube?

**Comment by MrYsLab** [comment]
@PG-InGitHub 
Thanks for the question. The LCD1602 typically comes with an i2c module. I am assuming that you are using that module to drive the LCD1602. If you are unfamiliar with i2c, it includes a specification for a serial protocol to communicate with i2c devices. All i2c devices use this protocol. Telemetrix provides an API to allow you to read and write to an i2c device, and it handles the protocol.
To control an i2c device, you need to read from and write data to the i2c device registers. The definition of the registers and how to use them differs for each i2c device. 
To use Telemetrix for an i2c device, one needs to translate the i2c code that typically resides on a board such as an Arduino to Telemetrix API calls. It is probably simpler to understand if you look at an example. [Here is one](https://github.com/MrYsLab/telemetrix/blob/master/examples/i2c_adxl345_accelerometer.py) for an ADXL345 accelerometer.
To put it simply, Telemetrix does not know the specifics about an i2c device's registers. It only knows how to transfer data to and from the device. To control the device, you need to consult its datasheet to understand its registers or translate the low-level i2c calls specific to the board you are using into the Telmetrix calls.

I am not very familiar with how Nixie tubes are controlled. If there is an i2c driver, you would need to do a similar translation into Telemetrix calls as mentioned above. I believe that some Nixie tube drivers use the SPI protocol. Telemetrix does support SPI, but only the data transport. You would need to understand the device's specifics in use and do the translation. Here is an[ example](https://github.com/MrYsLab/telemetrix/blob/master/examples/spi_mpu9250.py) using SPI to read an MPU9250 accelerometer.

If you have any questions, I would be happy to try and answer them.
